pkg/scheduler/scheduling_application.go

```diff
@@ -183,6 +183,23 @@ func (sa *SchedulingApplication) removeAllocationAsk(allocKey string) int {
 			delete(sa.requests, allocKey)
 		}
 	}
+	// Check if we need to change state based on the ask removal:
+	// 1) if pending is zero (no more asks left)
+	// 2) if allocation is zero (nothing is running)
+	// Change the state to waiting
+	if resources.IsZero(sa.pending) && resources.IsZero(sa.GetAllocatedResource()) {
+		oldState := sa.ApplicationInfo.GetApplicationState()
+		if err := sa.ApplicationInfo.HandleApplicationEvent(cache.WaitApplication); err != nil {
+			log.Logger().Warn("Application state not changed to Waiting while removing ask(s)",
+				zap.String("oldState", oldState),
+				zap.Int("releaseAllocs", toRelease),
+				zap.Error(err))
+		} else {
+			log.Logger().Debug("Application state change to Waiting while removing ask(s)",
+				zap.String("oldState", oldState),
+				zap.Int("releaseAllocs", toRelease))
```

Comment:
dup logging? we already did the logging for each time when app state changes

Reply:
yes I overlooked that one